keras/layers/preprocessing/audio_preprocessing.py

```diff
@@ -0,0 +1,548 @@
+import math
+
+from keras import ops
+from keras.api_export import keras_export
+from keras.layers.layer import Layer
+
+
+def gcd(a, b):
+    """Returns the greatest common divisor via Euclid's algorithm.
+
+    Args:
+        a: The dividend. A scalar integer `Tensor`.
+        b: The divisor. A scalar integer `Tensor`.
+
+    Returns:
+        A scalar `Tensor` representing the greatest common divisor
+        between `a` and `b`.
+
+    Raises:
+        ValueError: If `a` or `b` are not scalar integers.
+    """
+    a = ops.convert_to_tensor(a)
+    b = ops.convert_to_tensor(b)
+
+    a, b = ops.while_loop(
+        lambda _, b: ops.greater(b, ops.zeros_like(b)),
+        lambda a, b: [b, ops.mod(a, b)],
+        [a, b],
+    )
+    return a
+
+
+def frame(
```

Comment:
Hi @awsaf49 
I believe we already have `ops.math.extract_sequences` to handle the framing.
We should be able to extend the current API to support `pad_end`, `pad_value` and `axis` if needed.

https://github.com/keras-team/keras/blob/master/keras/ops/math.py#L277

Reply:
Thanks, I'll check it out too

Reply:
@james77777778 thanks for the suggestion, I've replaced `frame` with `ops.math.extract_sequences` as for melspec layer I don't need the `pad` option. Feel free to extend it for padding.